js/lex.go

```diff
@@ -66,21 +90,34 @@ func (tt TokenType) String() string {
 
 // Lexer is the state for the lexer.
 type Lexer struct {
-	r *buffer.Lexer
-
-	regexpState   bool
-	templateState bool
+	r     *buffer.Lexer
+	stack []ParsingContext
```

Comment:
I'd prefer not to conflate names with different meanings into one.

`enterContext` pushing a `context` onto `l.context` would be rather confusing for readability. `stack` (or `contextStack`) is not an implementation detail, but conveys what it actually is (that it's a FILO list of actual contexts, so you know in which order to expect them etc.).

> `ParsingContext` because it is pretty obvious that it is for parsing purposes as that is what the package is called.

Yeah but the class itself is `Lexer` so some disambiguation is useful.